### PR TITLE
Added support for GEOSEARCH and GEOSEARCHSTORE command + Claude related specs

### PR DESCRIPTION
**Pull request by vladvildanov**
Added one of the missing command API mentioned in #978. Also added Claude related context + custom skills for adding new commands. Also added a general spec for adding new commands best practices that both users and AI agents would follow to simplify contributions

---

> [!NOTE]
> **Low Risk**
> Additive geo command API and tests; Distributed multi-key routing for GEOSEARCHSTORE is the main behavioral edge case, with no auth or data-migration impact.
> 
> **Overview**
> Adds **Redis 6.2+** client support for **`GEOSEARCH`** and **`GEOSEARCHSTORE`** via keyword options (`frommember` / `fromlonlat`, `byradius` / `bybox`, sort, count/`count_any`, `WITH*` flags, `storedist`), reusing an extended **`_geoarguments`** helper (including **`COUNT … ANY`**).
> 
> **`Redis::Distributed`** gains geo wrappers (including **`geosearchstore`** with **`ensure_same_node`** / **`CannotDistribute`** when destination and source shard apart). Integration tests cover standalone, cluster (tagged keys), and distributed routing.
> 
> Also ships contributor/AI workflow docs (**`CLAUDE.md`**, **`specs/adding-commands.md`**, Claude **add-new-command** skill), ignores **`.bundle/`**, and hardens cluster tests (**replica client warmup**, **`with_acl`** on all nodes).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 84cd4713794dc04c47e6eeca0a82185e412fdc8c. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: lib/redis/commands/geo.rb [hunk 1]
```diff
@@ -61,6 +61,51 @@ def geopos(key, member)
...
+      # @param [Boolean] withhash also return the raw geohash-encoded sorted set score of the item
+      # @return [Array<String>] may be changed with WITH* flags
+      def geosearch(key, frommember: nil, fromlonlat: nil, byradius: nil, bybox: nil,
+                    sort: nil, count: nil, count_any: false,
+                    withcoord: false, withdist: false, withhash: false)
+        args = [key]
+        args << "FROMMEMBER" << frommember if frommember
+        args << "FROMLONLAT" << fromlonlat[0] << fromlonlat[1] if fromlonlat
+        args << "BYRADIUS" << byradius[0] << byradius[1] if byradius
+        args << "BYBOX" << bybox[0] << bybox[1] << bybox[2] if bybox
```
[the remark this answers is not in the document]
Comment (vladvildanov) [reply lib/redis/commands/geo.rb:97]: This is aligned with project existing behaviour

File: lib/redis/commands/geo.rb [hunk 2]
```diff
@@ -73,10 +118,13 @@ def geodist(key, member1, member2, unit = 'm')
...
-      def _geoarguments(*args, options: nil, sort: nil, count: nil)
+      def _geoarguments(*args, options: nil, sort: nil, count: nil, count_any: false)
         args << sort if sort
-        args << 'COUNT' << Integer(count) if count
-        args << options if options
+        if count
+          args << 'COUNT' << Integer(count)
+          args << 'ANY' if count_any
+        end
+        args.concat(Array(options))
```
[the remark this answers is not in the document]
Comment (vladvildanov) [reply lib/redis/commands/geo.rb:169]: This is geo module specific behaviour, count_any without count wouldn't have any effect

**Review by petyaslavova — COMMENTED** [review]
Over all looks good. My only note is about the added md files: referencing concrete line numbers isn't ideal, since those lines can become obsolete quickly.

**Comment by vladvildanov** [comment]
@petyaslavova Good point, I fixed those
